### PR TITLE
Same site cookies

### PR DESCRIPTION
**Pull request by mhennoch**
- Depending if script loaded in iframe add SameSite attribute to cookie

### CODE REVIEW

File: integration-tests/tests/cookies/cookies.ejs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<head>
+	<meta charset="UTF-8">
+	<title>Cookies</title>
+
+	<%- renderAgent({debug: true}) %>
+</head>
+<body>
+	<h1>COOKIES</h1>
+
+  <iframe src="/cookies/iframe.ejs"></iframe>
```
Comment (jtmalinowski) [inline integration-tests/tests/cookies/cookies.ejs:12]: does it matter that the iframe has the same origin? I'd bet it does

File: src/main.js [hunk 2]
```diff
@@ -68,16 +68,27 @@ if (!window.SplunkRum) {
...
-    let rumSessionId = undefined;
+    let rumSessionId = instanceId;
     const cookieSetter = function() {
       if (!document.cookie.includes(cookieName)) {
         const sessionId = generateId(128);
-        document.cookie = cookieName + '=' + sessionId + '; path=/; max-age=' + SessionTimeoutSeconds ;
+        let cookie = cookieName + '=' + sessionId + '; path=/; max-age=' + SessionTimeoutSeconds ;
+
+        if (isIframe()) {
+          cookie += ';SameSite=None; Secure';
```
Comment (jtmalinowski) [inline src/main.js:81]: How about we start building the decision doc in an .md file? Would be great to add explanations and reasoning behind all new decision together with the code to implement them.

File: src/utils.js [hunk 3]
```diff
@@ -40,3 +40,11 @@ export function limitLen(s, cap) {
     return s;
   }
 }
+
+export function isIframe() {
```
Comment (jtmalinowski) [inline src/utils.js:44]: I can see usefulness of exposing this and testing e2e across browsers, that said, I don't know what's the propensity of this to break, so you decide Marten, because you did the research.

File: src/main.js (hunk 2 above)
Comment (mhennoch) [reply src/main.js:81]: Yep, good idea, I've thought about it also.

File: src/utils.js (hunk 3 above)
Comment (mhennoch) [reply src/utils.js:44]: Supposedly this is the safest and most widely supported way according to stackoverflow :P. 
Integration tests should cover this. Even if this breaks nothing would break for our users.

File: integration-tests/tests/cookies/cookies.ejs (hunk 1 above)
Comment (mhennoch) [reply integration-tests/tests/cookies/cookies.ejs:12]: It does but we don't have a good way of testing third party iframe. I could start another http server on another port or use some third party  external page somehow. As this is a corner case anyways the test only checks if correct cookie string is used in iframe vs non iframe.

Comment (johnbley) [reply integration-tests/tests/cookies/cookies.ejs:12]: Having the capability to serve from a second port would be awesome for, e.g., testing CORS scenarios.  Let's throw that on the backlog but I think what you have here is fine for now.

File: src/main.js (hunk 2 above)
Comment (johnbley) [reply src/main.js:81]: 👍

**Review by johnbley — APPROVED** [review]
